src/cacheExchange.ts

```diff
@@ -103,6 +120,13 @@ export const cacheExchange = (opts?: CacheExchangeOpts): Exchange => ({
     opts.keys
   );
 
+  let hydration: void | Promise<void>;
+  if (opts.storage) {
+    hydration = opts.storage.read().then(data => {
+      store.hydrateData(data, (opts as any).storage);
```

Comment:
What’s going on here with `any`?

Reply:
Typescript being weird.... It's saying `.storage` is possibly undefined....

Reply:
Seems to be because `storage` isn’t reassigned. Maybe we can try `opts.storage!`?

Reply:
This works.... `opts!.storage!`